### ISSUE TITLE
Incorrect entry in load-path for ess package

### ISSUE THREAD

**Issue by doolio** [raxod502/straight.el#498]
Firstly, I've just switched to `straight` and was impressed how painless it was. So well done.  

I install and configure `ess` via my `init.el` as follows:

```
(use-package ess-site                   ; ESS - Emacs Speaks Statistics
  :straight ess
  :commands R
  :hook (ess-mode . subword-mode))
```

The `straight` build as far as I can tell is successful however I see the following entries in my `load-path`. 
```
"/home/doolio/.emacs.d/straight/build/ess"
"/home/doolio/.emacs.d/straight/build/ess/"
"/home/doolio/.emacs.d/straight/build/ess/obsolete/"
```
The first and third entries make sense to me looking at the `repos/ESS` directory but I don't understand why I see the second entry. Now there is an `etc/` directory within my `build/ess` directory. Should this second entry instead be 

`"/home/doolio/.emacs.d/straight/build/ess/etc/"`

Thanks for your time.

Edit: correct path

**Comment by raxod502** [comment]
The first entry is standard, and is added by `straight.el`. The second entry is _not_ standard, and is added by ESS when you load it (see https://github.com/emacs-ess/ESS/blob/625041ad51038e43ac12e466e3001a252725d88b/lisp/ess-site.el#L56 and https://github.com/emacs-ess/ESS/blob/625041ad51038e43ac12e466e3001a252725d88b/lisp/ess-custom.el#L158). The third entry is nonstandard, and is added by ESS (see https://github.com/emacs-ess/ESS/blob/625041ad51038e43ac12e466e3001a252725d88b/lisp/ess-custom.el#L162).

In other words, the unexpected behavior is caused by nonstandard packaging. `straight.el` always adds one and only one entry to `load-path`, and expects that all Lisp files should be symlinked into that directory by the package's build spec.

**Comment by doolio** [comment]
I see. Thanks for the explanation. This likely is the reason for a similar issue  (https://github.com/rswgnu/hyperbole/issues/23) I reported with the hyperbole package and which still exists with straight when using your recommended receipe given in #237.

I'm happy to report this issue upstream but unsure how I should word it if you have any suggestions. In any case, feel free to close this issue. Thanks for your time.

**Comment by raxod502** [comment]
I do not know if upstream will be amenable to modernizing their packaging. Large packages do seem to be attracted to nonstandard packaging for some reason. But, if you want to give it a go, I'd point out that if ESS is installed using any package manager, then it will already be added to the `load-path`, and therefore [this line](https://github.com/emacs-ess/ESS/blob/625041ad51038e43ac12e466e3001a252725d88b/lisp/ess-custom.el#L158) can be removed. I think the line in `ess-site.el` is reasonable, because if I understand correctly, that file shouldn't be loaded unless you are trying to install ESS manually outside of a package manager. And finally the line having to do with `obsolete` also seems reasonable, because Emacs' feature of producing a warning when you load a file inside a directory called `obsolete` won't work if the `obsolete` files aren't actually in a subdirectory called `obsolete` (so therefore a `load-path` entry needs to be added).

**Comment by doolio** [comment]
Thanks. I've opened [ESS/issues/1000](https://github.com/emacs-ess/ESS/issues/1000) upstream. Hopefully, it explains the issue sufficiently and that they are amenable to the suggested correction. In any case, I think you can close this issue. Thanks again for your time.

**Comment by doolio** [comment]
Upstream were very quick to respond and am happy to report with their fix I now only have the following entries in my `load-path`:
```
"/home/doolio/.emacs.d/straight/build/ess"
"/home/doolio/.emacs.d/straight/build/ess/obsolete/"
```
Therefore, this issue can be closed. Thanks again.
